src/networking/network.rs

```diff
@@ -299,6 +300,23 @@ impl Network {
             }
         }
     }
+
+    // For transaction propagation made by mempool to all peers
+    async fn propagate_transaction_to_peers(tx: Transaction) {
+        let peers_db_global = PEERS_DB_GLOBAL.clone();
+        let mut peers_db_mut = peers_db_global.write().await;
+        // We need a stream iterator for async(to await send_command_fire_and_forget)
+        let mut peers_iterator_stream = futures::stream::iter(peers_db_mut.values_mut());
+        while let Some(peer) = peers_iterator_stream.next().await {
+            if peer.get_has_completed_handshake() && !peer.is_in_path(tx.clone()) {
```

Comment:
Let's try to get rid of the clone() here... It shouldn't be needed, we are only reading the tx, right?